src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -91,22 +91,22 @@ var MarkdownHighlightRules = function() {
     // handle highlighting for *abc*, _abc_ separately, as pandoc's
     // parser is a bit more strict about where '_' can appear
     var strongUnderscore = {
-        token: ["text", "constant.numeric"],
+        token: ["text", "constant.numeric.text"],
```

Comment:
I was wondering about the "text" type in the first index of the array here, it doesn't seem to carry over to the spellchecker so I did have to add the ".text" in the second array parameter. If it's ok/preferred to delete the first "text" just let me know.

Reply:
For regular expressions that include captures, Ace will map the types in the array back to the capture (by index) In this case:

```
        token: ["text", "constant.language.boolean.text"],
        regex: "(\\s+|^)(_(?=[^\\s_])(?:(?:\\\\.)|(?:[^\\s_\\\\]))*?_)\\b"
                 ^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                  text    constant.language.boolean.text
```

You'll notice that some of the type names are a bit bogus (why is the text within underscores given a 'boolean' class !?) but this is mostly us just being lazy and using that type because it maps to some CSS style in our existing set of themes that we wanted to apply here.